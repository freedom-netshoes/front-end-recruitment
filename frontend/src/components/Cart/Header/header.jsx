import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import { connect } from 'react-redux'
import CSSModules from 'react-css-modules'
import { toggleCart } from '../../../actions/action'
import ButtonDismiss from '../../crossComponents/ButtonDismiss/button-dismiss'
import styles from './header.less'

@CSSModules(styles)
class Header extends Component {
  static defaultProps = {
    size: 0
  };

  static propTypes = {
    size: PropTypes.number
  };

  render () {
    return (
      <h1 className='cart-title'>
        <ButtonDismiss {...{
          clickButton: () => {
            this.props.dispatch(toggleCart(false))
          }
        }}
        />
        <span styleName='bag'>
          <span styleName='total-items'>{this.props.size}</span>
        </span>
        Sacola
      </h1>
    )
  }
}

export default connect()(Header)
