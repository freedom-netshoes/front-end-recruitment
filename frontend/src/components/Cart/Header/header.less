@import url('../../../styles/_less-mixins.less');
:global {
    .cart-title {
        .fontBold();
        color: @white;
        font-size: 24px;
        margin: 0px;
        padding: 48px 0;
        text-transform: uppercase;
    }
}

.bag {
    background: url('\./bag.png') 0 0 no-repeat;
    display: inline-block;
    height: 40px;
    margin-right: 20px;
    position: relative;
    top: 10px;
    width: 33px;
}

.total-items {
    .fontBold();
    background-color: @gold;
    border-radius: 50%;
    bottom: -8px;
    color: @black;
    font-size: 13.8px;
    height: 24px;
    line-height: 23px;
    position: absolute;
    right: -8px;
    text-align: center;
    width: 24px;
}

:global {
    .cart-title {
        .button {
            position: absolute;
						left: 20px;
            top: 10px;
        }
    }
}