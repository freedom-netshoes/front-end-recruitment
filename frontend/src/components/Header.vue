<template>
  <div class="header">
    <h3>Tiago Juvenal de Lima</h3>
  </div>
</template>

<script>
export default {
  name: 'CcHeader'
}
</script>

<style lang="scss" scoped>
  .header {
    width: 100%;
    background: #202025;

    h3 {
      font-size: 1vw;
      text-align: center;
      color: white;
      padding: 2% 0;
      font-weight: bold;
    }
  }
</style>
