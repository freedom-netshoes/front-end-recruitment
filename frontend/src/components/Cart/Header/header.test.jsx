/* global describe, it, expect */
import React from 'react'
import Header from './header'
import { shallow } from 'enzyme'
import getStore from '../../../resources/store'

describe('Testing <Header /> component', () => {
  it('renders the Header component', () => {
    const component = shallow(<Header store={getStore()} />)
    expect(component).toHaveLength(1)
  })
  it('should test snapshop in <Header >', () => {
    let component = shallow(<Header store={getStore()} size={10} />)
    expect(component).toMatchSnapshot()
  })
  // it('should button to be clicked', () => {
  //   const component = shallow(<Header store={getStore()} size={10} />)
  //   expect(component).toMatchSnapshot()
  //   component.find('ButtonDismiss').simulate('click')
  //   expect(component).toMatchSnapshot()
  // })
})
