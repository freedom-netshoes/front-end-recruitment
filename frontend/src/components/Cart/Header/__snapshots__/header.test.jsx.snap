// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Header /> component should test snapshop in <Header > 1`] = `
<Header
  dispatch={[Function]}
  size={10}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
